slang-testing/src/main/java/com/sonarsource/slang/testing/TreeAssert.java

```diff
@@ -97,6 +97,24 @@ public TreeAssert hasChildren(Class... classes) {
     return this;
   }
 
+  public TreeAssert hasDescendant(Tree expected) {
```

Comment:
Refactor this method and hasNotDescendant using an auxiliary method that returns whether a tree is a descendant. That logic is replicated in both methods and it could be good to isolate in case it changes in the future.